.github/workflows/auto-merge.yml

```diff
@@ -0,0 +1,66 @@
+name: Dependabot PR Handling
+
+on:
+    pull_request:
+      types: [opened, reopened]
+      branches:
+          - master
+          - main
+
+jobs:
+  check-dependabot:
+    runs-on: ubuntu-20.04
+    outputs:
+      is_dependabot_pr: ${{ steps.dependabot-check.outputs.is_dependabot_pr }}
+    steps:
+      - id: dependabot-check
+        name: Check if PR is from Dependabot
+        run: |
+          echo "PR author: ${{ github.event.pull_request.user.login }}"
+          if [[ "${{ github.event.pull_request.user.login }}" == "dependabot[bot]" || "${{ github.event.pull_request.user.login }}" == "dependabot-preview[bot]" ]]; then
+            echo "::set-output name=is_dependabot_pr::true"
+          else
+            echo "::set-output name=is_dependabot_pr::false"
+          fi
+
+  tests:
+    needs: check-dependabot
+    runs-on: ubuntu-20.04
+    if: ${{ needs.check-dependabot.outputs.is_dependabot_pr == 'true' }}
```

Comment:
We might make changes manually ourselves as well, will the tests not run for those changes? Maybe we can run the test job first always and then run the dependabot as subsequent steps if the tests succeeded wdyt